content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+
+## ExtractResource Trait and Plugin
+
+<div class="release-feature-authors">authors: Rob Swain (@superdump)</div>
+
+Some ECS resources have very simple extract logic:
+
+```rust
+fn extract_cool_color(mut extracted_time: ResMut<CoolColor>, time: Extract<Res<CoolColor>>) {
+    *cool_color = cool_color.clone();
+}
+```
+
+Rather than force developers to write this out manually, **Bevy 0.8** now provides the [`ExtractResource`] trait / derive:
+
+```rust
+#[derive(ExtractResource, Clone)]
+struct CoolColor {
+    color: Color,
+}
+```
+
+Then, just add the [`ExtractResourcePlugin<CoolColor>`][`ExtractResourcePlugin`] to your [`App`] and the resource will be automatically extracted.
+
+[`ExtractResource`] can also be implemented manually if you need custom logic (or the type needs to change):
+
+```rust
+impl ExtactResource for ExtractedCoolColor {
+    type Source = CoolColor;
+    fn extract_resource(source: &CoolColor) -> Self {
+        Self {
+            color: source.color.as_rgba_linear(),
+        }
+    }
+}
+```
+
+
+[`ExtractResource`]: https://docs.rs/bevy/0.8.0/bevy/render/extract_resource/trait.ExtractResource.html
+[`ExtractResourcePlugin`]: https://docs.rs/bevy/0.8.0/bevy/render/extract_resource/struct.ExtractResourcePlugin.html
+[`App`]: https://docs.rs/bevy/0.8.0/bevy/app/struct.App.html
+
+## Query IntoIter
+
+<div class="release-feature-authors">authors: @TheRawMeatball</div>
+
+Bevy ECS [`Queries`][`Query`] now implement the [`IntoIterator`] trait, which provides access to Rust's nicer iteration syntax:
+
+```rust
+// In previous Bevy versions, iter/iter_mut had to be called manually:
+fn system(mut players: Query<&mut Player>) {
+    for player in players.iter() {
+    }
+    
+    for mut player in players.iter_mut() {
+    }
+}
+
+// In Bevy 0.8, you have the option to use this syntax:
+fn system(mut players: Query<&mut Player>) {
+    for player in &players {
+    }
+    
+    for mut player in &mut players {
+    }
+}
+```
+
+[`IntoIterator`]: https://doc.rust-lang.org/std/iter/trait.IntoIterator.html
+[`Query`]: https://docs.rs/bevy/0.8.0/bevy/ecs/system/struct.Query.html
+
+## Query::iter_many
+
+<div class="release-feature-authors">authors: @devil-ira</div>
+
+It is now possible to pass a list of entities to a [`Query`] to iterate over using [`Query::iter_many`], which is significantly more ergonomic than calling `get(entity)` on each individual entity:
+
+```rust
+#[derive(Component)]
+struct BlueTeam {
+    members: Vec<Entity>,
+}
+
+fn system(blue_team: Res<BlueTeam>, players: Query<&Player>) {
+    info!("Blue Team Assemble!");
+    for player in players.iter_many(&blue_team.members) {
+        info!("{}", player.name);
+    }    
+}
+```
+
+There is also a [`Query::many_for_each_mut`], which provides similar functionality for mutable queries, but it uses a closure to ensure there is never aliased mutability if the given list contains duplicates:
```

Comment:
We could merge https://github.com/bevyengine/bevy/pull/5402 and make the API cleaner from the start if you wanted.